src/app/components/MovieButton/MovieButton.tsx

```diff
@@ -2,47 +2,52 @@ import React from 'react';
 import BookmarkIcon from '../assets/BookmarkIcon';
 import HomeIcon from '../assets/HomeIcon';
 import SearchIcon from '../assets/SearchIcon';
-import StarIcon from '../assets/StarIcon';
 
-type MovieButtonProps = {
-  iconType:
-    | 'searchActive'
-    | 'searchInactive'
-    | 'homeActive'
-    | 'homeInactive'
-    | 'starActive'
-    | 'starHalfActive'
-    | 'starInactive'
-    | 'bookmarkActive'
-    | 'bookmarkInactive';
+export type MovieButtonProps = {
+  iconType: 'search' | 'home' | 'bookmark';
+  onHandleButtonClick: () => void;
+  toggleActiveState: () => void;
+  isActive: boolean;
 };
 
-const MovieButton = ({ iconType }: MovieButtonProps): JSX.Element => {
+const MovieButton = ({
+  isActive,
+  iconType,
+  onHandleButtonClick,
+  toggleActiveState,
+}: MovieButtonProps): JSX.Element => {
+  const active = {
+    fill: 'var(--color-primary)',
+    stroke: 'var(--color-primary)',
+  };
+
+  const inactive = {
+    fill: iconType !== 'bookmark' ? 'var(--text-primary)' : 'transparent',
+    stroke: 'var(--text-primary)',
+  };
+
+  const getProps = () => {
+    return isActive ? active : inactive;
+  };
```

Comment:
You can just add this logic in place of where this function is called and spread what the ternary returns